README.md

```diff
@@ -66,102 +66,190 @@ Server-side rendering of Riot Components using the [_Vision_](https://github.com
+});
+```
 
-### `removeCache`
+> Note above uses [inert](https://github.com/hapijs/inert) to serve up static files.
 
-While you are _developing_ your app you typically don't want
-your views to be _cached_, however, when you deploy your app
-by setting `NODE_ENV=production` views will be cached.
+### 4. Configure Vision Plugin with Hapi-Riot
 
-If for _any_ reason you _want_ to cache your views during development,
-or if you need to debug production on your localhost,
-set the `compileOptions.removeCache: false` like this:
+You can configure the views engine by passing it `compileOptions`
 
 ```js
 server.views({ // initialise
   engines: {
     tag: require('hapi-riot') // file should be .tag
   },
-  relativeTo: __dirname, // assuming your
+  relativeTo: __dirname,
   path: 'views', // by convention we put our tags in /views dir
-  compileOptions: {
-    removeCache: false
+  compileOptions: { // of this form, see below for meanings
+    removeCache,
+    compiledFileRoute, // REQUIRED same as step 3
+    layoutPath,
+    layout
   }
 });
+
 ```
 
+We have added a few _features_ to _simplify_ our own projects but are not meant to be fit for _everyone_.
 
 
-## Lessons Learned & "Gotchas"
+#### `compiledFileRoute` **REQUIRED**
 
-### 1. Don't leave tag files empty!
+This is the oath you specified in step 3 to include javascript within your tags. `hapi-riot`
+will then inject a link to your compiled file into your output which makes the
+specified methods in your tags available.
 
-e.g: if you create a new tag `views/bye.tag` and _forget_ to
-write some `html` code inside it you will see a _stranage_ error
-when you attempt to render it or _any other existing tag_!
+#### `removeCache` *default: `process.env.NODE_ENV === 'production'`*
+
+While you are _developing_ your app you typically don't want
+your views to be _cached_, however, when you deploy your app
+by setting `NODE_ENV=production` views will be cached. If for _any_ reason you _want_ to cache your views during development set the value as true;
+
+#### `layoutPath` *REQUIRED if layout defined*
+
+The place where your layouts are kept.
+
+> Note: path does not make use of the relativeTo param given to vision plugin
+
+#### `layout` *default: undefined*
+
+Can either be set to `true` in which case we will look for a `layout.html` file or you can specify which ever file you'd like. Can be overridden from `reply.view` to have multiple layouts.
+
+Specifying a layout allows you to provide a core html page for your riot content where you can include style sheets, other html content and other base scripts. We will inject the riot content into a place holder `<<<RIOT>>>` which much be present in the file.
+
+An example would be
 
-When you create a new tag add _something_ to it immediately. e.g: `views/bye.tag`
-```js
-<bye>
-  <h1> Goodbye {opts.name}! </h1>
-</bye>
 ```
+<html>
+  <head>
+    <link rel="stylesheet" type="text/css" href="stylesheet.css"/>
+  <head>
+  <body>
+    <<<RIOT>>>
```

Comment:
@nelsonic do you mean that we should replace our `layout.html` file with a riot `.tag` file that yields to our content?